File: bmh_parent/bmh_rope_web/src/main/webapp/js/service/clientService.js
//服务层
app.service('clientService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('./client/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('./client/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('./client/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('./client/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('./client/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('./client/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('./client/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: bmh_parent/bmh_rope_interface/src/main/java/com/bmh/rope/service/ClientService.java
package com.bmh.rope.service;

import com.bmh.pojo.Client;
import entity.PageResult;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ClientService {

    /**
     * 返回全部列表
     * @return
     */
    public List<Client> findAll();


    /**
     * 返回分页列表
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);


    /**
     * 增加
     */
    public void add(Client seller);


    /**
     * 修改
     */
    public void update(Client seller);


    /**
     * 根据ID获取实体
     * @param id
     * @return
     */
    public Client findOne(String id);


    /**
     * 批量删除
     * @param ids
     */
    public void delete(String[] ids);

    /**
     * 分页
     * @param pageNum 当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(Client seller, int pageNum, int pageSize);

    void updateStatus(String status, String sellerId);
}
